ArrayOrder.md

```diff
@@ -0,0 +1,202 @@
+## Multi-dimensional array indexing
+
+Zarr stores multi-dimensional arrays into regularly sized chunks.
+Chunks are themselves multi-dimensional arrays of a smaller size than
+the complete multidimensional array and are stored as a 1D array of
+values, called a "flattened" array.
+
+Indexing into a multi-dimensional array is done with an ordered tuple,
+each element of which indexes into one of the array's dimensions.
+Elements of this tuple are called "coordinates." For example, the tuple
+`(i,j,k)` indexes a three-dimensional array, and `i`, `j`, and `k` are
+its coordinates. We will call `i` the "left" or "first" index, and `k` the
+"right" or "last" index.
+
+The only valid coordinates for arrays are the non-negative integers.
```

Comment:
What's the argument for claiming that negative numbers are "invalid" indices? tensorstore allows indexing with negative numbers: https://google.github.io/tensorstore/python/indexing.html#integer-indexing.

Reply:
"invalid" for the purpose of this document / discussion. not generally